### PR TITLE
RoR Blog Web App: Milestone-2 Processing dta

### PR DESCRIPTION
**Pull request by Profsain**
### I completed this part of the project by fulfilling the requirement below:

- Created models
     Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
    Set up associations between models.
- Use models to insert data
- Created custom methods
A method that returns the 3 most recent posts for a given user.
A method that updates the posts counter for a user.
A method which returns the 5 most recent comments for a given post.
A method that updates the comments counter for a post.
A method that updates the likes counter for a post.
Go to rails c and check if your methods are working.

### CODE REVIEW

File: app/models/author.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
+class Author < ApplicationRecord
+end
```
Comment (Mihndim2020) [inline app/models/author.rb:4]: - [ ] As it is a requirement for this project, please ensure associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Kindly create an association between user, posts, comments and likes. Something like so: 
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

- [ ] According to the project requirements, User model should have a  method that returns 3 most recent posts. Please kindly implement that. Something like so: 

```ruby
def recent_post
  posts.order('created_at Desc').limit(3)
end
```

File: app/models/post.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author
```
Comment (Mihndim2020) [inline app/models/post.rb:4]: - [ ] As it is a requirement for this project, please ensure associations between models are set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Kindly create an association between posts, user, comments and likes. Something like so: 
```ruby
belongs_to :author, class_name: 'User'
has_many :comments
has_many :likes
```

- [ ] According to the project requirements, the Post model should have a method that updates the posts counter for a user. Something like so: 
```ruby
def update_post_counter
  author.increment!(:posts_counter)
end
```
Please kindly fix that.

- [ ] According to the project requirements, the Post model should have a method that returns the 5 most recent comments for a given post. Something like so: 
```ruby
def recent_comments
  comments.order('created_at Desc').limit(5)
end
```

Please kindly fix that.

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
 class Comment < ApplicationRecord
-  blongs_to :user
+  belongs_to :author
   belongs_to :post
```
Comment (Mihndim2020) [inline app/models/comment.rb:5]: - [ ]  As it is a requirement for this project, the Comment model should have a method that updates the comments counter for a post.  Something like so: 
 ```ruby
def update_comments_counter
  post.increment!(:comments_counter)
end
```
Please kindly fix that.

File: app/models/like.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
 class Like < ApplicationRecord
-  belongs_to :user
+  belongs_to :author
   belongs_to :post
```
Comment (Mihndim2020) [inline app/models/like.rb:5]: - [ ]  As it is a requirement for this project, the Like model should have a method that updates the likes counter for a post.  Something like so: 
 ```ruby
def update_likes_counter
  post.increment!(:likes_counter)
end
```
Please kindly fix that.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @Profsain, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/author.rb (hunk 1 above)
Comment (Profsain) [reply app/models/author.rb:4]: Thank you for your feedback.
I will implement the changes required.

File: app/models/comment.rb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
 class Comment < ApplicationRecord
-  blongs_to :user
+  belongs_to :author
```
Comment (tufoinnkuo10) [inline app/models/comment.rb:4]: - [ ] please like the previous reviewer noted, the Comment model should have a method that updates the comments counter for a post. Something like so:
```
def update_comments_counter
 post.increment!(:comments_counter)
end
```

File: app/models/post.rb [hunk 6]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'Author'
```
Comment (tufoinnkuo10) [inline app/models/post.rb:4]: - [ ] pleasr like the previous reviewer noted,the Post model should have a method that updates the posts counter for a user. Something like so:
```
def update_post_counter
  author.increment!(:posts_counter)
end
```

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @Profsain 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated


Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**